README.md

```diff
@@ -1,50 +1,96 @@
-# Universal Permissionless Rewards Distributor
+# Universal Rewards Distributor
+The Universal Rewards Distributor (URD), is a smart contract that allows for the distribution of multiple ERC20 tokens from a single off-chain computed Merkle tree.
 
-A universal permissionless rewards distributor written in Solidity. It allows the distribution of any reward token (different reward tokens are possible simultaneously) based on a Merkle tree distribution, and using ERC20 allowance of a treasury.
+Each URD contract has an owner and a group of updaters. These updaters can submit a new root to a pending value, which is subject to a timelock (that can be set to 0). Once the timelock period ends, anyone can accept this pending value. However, during the timelock period, the owner has veto power and can remove the pending value.
 
-The singleton contract allows any treasury to distribute rewards to any address, based on a Merkle tree. The Merkle root is stored in the contract. The Merkle root can be updated by whitelisted users. The distribution owner can freeze, force update, or suggest a new treasury. The treasury must accept the role and set an allowance to distribute rewards.
+## Use Case Example
 
-Based on [Morpho's rewards distributor](https://github.com/morpho-dao/morpho-v1/blob/main/src/common/rewards-distribution/RewardsDistributor.sol), itself based on [Euler's rewards distributor](https://github.com/euler-xyz/euler-contracts/blob/master/contracts/mining/EulDistributor.sol).
+- Assume I am a DAO with a periodic rewards mechanism. Each month, a [Gelato](https://www.gelato.network/) bot runs a script creating a Merkle tree that distributes TokenA and TokenB.
```

Comment:
```suggestion
- Assume that a DAO with a periodic rewards mechanism. Each month, a [Gelato](https://www.gelato.network/) bot runs a script creating a Merkle tree that distributes TokenA and TokenB.
```

Weird to use the first person, especially for a DAO

Reply:
User impersonating but I agree in some sense.

But your suggestion is not correct: "Assume that a DAO with a periodic rewards mechanism"